.github/workflows/snap-store-publish-to-candidate.yml

```diff
@@ -78,21 +82,78 @@ jobs:
         run: |
           rev=$(snapcraft push $SNAP_NAME --release=$CHANNEL)
           echo "revision_${ARCHITECTURE}=${rev}" >> $GITHUB_OUTPUT
-    outputs: 
-      revision_amd64: ${{ steps.publish.outputs.revision_amd64 }}
-      revision_arm64: ${{ steps.publish.outputs.revision_arm64 }}
-      version: ${{ steps.build.outputs.version }}
+
+      - name: Upload snap artifact for testing
+        uses: actions/upload-artifact@v3
+        with:
+          name: snap
+          path: ${{ steps.build.outputs.snap }}
+
+  # Create an issue using the template that asks maintainers to test the snap, and take
+  # action to promote the revisions to stable if testing is successful.
   create_issue:
     name: "Create call for testing"
     needs: build
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
       - uses: JasonEtco/create-an-issue@v2
+        id: comment
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           revision_amd64: ${{ needs.build.outputs.revision_amd64 }}
           revision_arm64: ${{ needs.build.outputs.revision_arm64 }}
           version: ${{ needs.build.outputs.version }}
         with:
           filename: .github/testing-issue-template.md
+    outputs:
+      issue_number: ${{ steps.comment.outputs.number }}
+
+  # Deploy the snap inside a desktop VM and grab screenshots of the desktop, and of
+  # the application, then post them as a comment on the issue created above.
+  grab-screenshots:
+    name: Gather screenshots
+    needs: 
+      - build
+      - create_issue
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v4
+
+      - name: Get the snap artifact
+        uses: actions/download-artifact@v3
```

Comment:
Switch to downloading the snap from the store, if at all possible. The current tokens should allow you to download an arbitrary snap revision, so we don't have to worry about race conditions where the revision-to-test isn't the latest revision anymore.